pkg/action/install.go

```diff
@@ -111,6 +111,49 @@ func (i *Install) Run(chrt *chart.Chart, vals map[string]interface{}) (*release.
 		return nil, err
 	}
 
+	// Pre-install anything in the crd/ directory. We do this before Helm
+	// contacts the upstream server and builds the capabilities object.
+	if crds := chrt.CRDs(); !i.ClientOnly && !i.SkipCRDs && len(crds) > 0 {
+		// We do these one at a time in the order they were read.
+		for _, obj := range crds {
+			// Read in the resources
+			res, err := i.cfg.KubeClient.Build(bytes.NewBuffer(obj.Data))
+			if err != nil {
+				// We bail out immediately
+				return nil, errors.Wrapf(err, "failed to install CRD %s", obj.Name)
+			}
+			// On dry run, bail here
+			if i.DryRun {
+				i.cfg.Log("WARNING: This chart or one of its subcharts contains CRDs. Rendering may fail or contain inaccuracies.")
+				continue
+			}
+			// Send them to Kube
+			if _, err := i.cfg.KubeClient.Create(res); err != nil {
+				// If the error is CRD already exists, continue.
+				if apierrors.IsAlreadyExists(err) {
+					crdName := res[0].Name
+					i.cfg.Log("CRD %s is already present. Skipping.", crdName)
+					continue
+				}
+				return nil, err
+			}
+
+			// Invalidate the local cache.
+			if discoveryClient, err := i.cfg.RESTClientGetter.ToDiscoveryClient(); err != nil {
+				// On error, we don't want to bail out, since this is unlikely
+				// to impact the majority of charts.
+				i.cfg.Log("Could not clear the discovery cache: %s", err)
+			} else {
+				i.cfg.Log("Clearing discovery cache")
+				discoveryClient.Invalidate()
+				// Give time for the CRD to be recognized.
+				time.Sleep(5 * time.Second)
```

Comment:
We should probably clean up and reuse the code found here in Helm 2 so we aren't just sleeping and hoping it works: https://github.com/helm/helm/blob/3ed073b2e0c37dd5c994562d6fe8e04b5f3609eb/pkg/kube/client.go#L584-L612

Reply:
I am adding support to Wait() for CRDs, and then just using that. It has some hidden advantages for if people try to sneak non-CRD things into the CRDs() directory. But _sssssh... don't tell them that they can do this_.

Reply:
![tenor-138232708](https://user-images.githubusercontent.com/4294228/64203015-d1da6580-ce4f-11e9-82f2-4c6487485beb.gif)

Reply:
@thomastaylor312 Okay, this is now done. We're using an updated version of the Wait logic to watch for the CRD to become available.